README.md

```diff
@@ -160,142 +161,194 @@ Structured Extraction is guided by the provided schema. We support Pydantic Mode
-    url = f"https://example.com/?number={value}"
-    req = urllib.request.Request(url, method="GET")
-    with urllib.request.urlopen(req) as response:
-        return response.read()
+from typing import Dict
+from openai import OpenAI
+from tensorlake.applications import application, function, run_local_application, Image
 
+image = (
+    Image(base_image="python:3.11-slim", name="openai_story_writer")
+    .run("pip install openai")
+)
 
-# Define the full workflow using Graph object
-g = Graph(
-    name="example_workflow",
-    start_node=generate_sequence,
-    description="Example workflow",
+@function(
+    description="Analyzes the sentiment of input text",
+    secrets=["OPENAI_API_KEY"],
+    image=image
 )
-g.add_edge(generate_sequence, squared)
-g.add_edge(squared, sum_all)
-g.add_edge(sum_all, send_to_web_service)
-
-# Invoke the workflow for sequence [0..200].
-def run_workflow(g: Graph) -> None:
-    invocation_id: str = g.run(last_sequence_number=200, block_until_done=True)
-
-    # Get the output of the the workflow (of its last step).
-    last_step_output: str = g.output(invocation_id, "send_to_web_service")
-    click.secho("Web service response:", fg="green", bold=True)
-    click.echo(last_step_output[0])
-    click.echo()
-
-    # Get the sum.
-    sum_output: str = g.output(invocation_id, "sum_all")
-    click.secho("Sum:", fg="green", bold=True)
-    click.echo(sum_output[0])
-    click.echo()
+def analyze_sentiment(feedback: str) -> str:
+    """Step 1: Analyze the sentiment of the input text."""
+    print(f"Analyzing: {feedback}")
+    client = OpenAI()
+    response = client.chat.completions.create(
+        model="gpt-4o",
+        messages=[
+            {"role": "system", "content": "You are a sentiment analyzer. Respond with only one word: POSITIVE or NEGATIVE."},
+            {"role": "user", "content": feedback}
+        ]
+    )
+    return response.choices[0].message.content.strip()
+
+@function(
+    description="Drafts a customer support email based on the sentiment.",
+    secrets=["OPENAI_API_KEY"],
+    image=image
+)
+def draft_response(sentiment: str) -> str:
+    """Step 2: Draft a customer support email based on the sentiment."""
+    print(f"Drafting email for {sentiment} feedback...")
+    if sentiment == "NEGATIVE":
+        prompt = "Write a short, empathetic apology email to a customer."
+    else:
+        prompt = "Write a short, enthusiastic thank you email to a customer."
+    
+    client = OpenAI()
+    response = client.chat.completions.create(
+        model="gpt-4o",
+        messages=[{"role": "user", "content": prompt}]
+    )
+    return response.choices[0].message.content
+
+@application(
+    tags={"type": "quickstart", "use_case": "customer_support"},
+)
+@function(
+    description="Customer Support application.",
+    image=image
+)
+def customer_support(feedbacks: Dict[str, str]) -> Dict[str, str]:
```

Comment:
Use NamedDict or Pydantic models here to make it clear what the object looks like